### PR TITLE
Embedded source

### PR DESCRIPTION
**Pull request by KevinRansom**
Add support for embedded source code files in pdbs.
Update command line to support /embed and /embed:
Add some more command line testing.

Do not support /embed in fsi.

### CODE REVIEW

**Comment by KevinRansom** [comment]
@dotnet-bot bot test this please

File: src/absil/ilwritepdb.fs [hunk 1]
```diff
@@ -19,6 +19,31 @@ open Microsoft.FSharp.Compiler.AbstractIL.Internal.Library
...
 open Microsoft.FSharp.Compiler.Range
 
+
+type BlobBuildingStream () =
+    inherit Stream()
+
+    static let chunkSize = 32 * 1024
+    let builder = new BlobBuilder(chunkSize)
+
+    override this.CanWrite  with get() = true
```
Comment (dsyme) [inline src/absil/ilwritepdb.fs]: "with get()" is not idiomatic - you can just remove it

File: src/absil/ilwritepdb.fs [hunk 2]
```diff
@@ -253,23 +278,71 @@ let generatePortablePdb fixupSPs showTimes (info:PdbData) =
...
+    ///
+    /// Chosen as the point at which we start to see > 10% blob size reduction using all
+    /// current source files in corefx and roslyn as sample data. 
+    /// </summary>
+    let sourceCompressionThreshold = 200
+
     let documentIndex =
+        let includeSource file =
+            let isInList =
+                if embedSourceList |> List.length = 0 then false
```
Comment (dsyme) [inline src/absil/ilwritepdb.fs]: You can just use ``.Length`` instead of ``|> List.length``

**Comment by dsyme** [comment]
Nice work :)

File: src/absil/ilwritepdb.fs (hunk 1 above)
Comment (KevinRansom) [inline src/absil/ilwritepdb.fs]: done

File: src/absil/ilwritepdb.fs (hunk 2 above)
Comment (KevinRansom) [inline src/absil/ilwritepdb.fs]: done

Comment (KevinRansom) [inline src/absil/ilwritepdb.fs]: done

**Pull request merged**